### PR TITLE
Polyglot framework

### PR DESCRIPTION
**Pull request by lowjoel**
Depend on the Coursemology polyglot gem for language definitions.

### CODE REVIEW

**Comment by lowjoel** [comment]
@allenwq any comments?

**Comment by allenwq** [comment]
I didn't get notifications from these repos ><

File: lib/coursemology/polyglot/extensions/language.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Coursemology::Polyglot::Language
+  # Finds the language class with the specified name.
+  #
+  # @param [String] type The name of the class.
+  # @return [nil] If the type is not defined.
+  # @return [Class] If the type was found.
+  def self.find_by(type:)
```
Comment (allenwq) [inline lib/coursemology/polyglot/extensions/language.rb]: Wah, you like meta programming so much !

Comment (lowjoel) [inline lib/coursemology/polyglot/extensions/language.rb]: Haha, to make the API look coherent...

**Comment by lowjoel** [comment]
Might have to go watch the repo then :/

**Pull request merged**